docs/pages/database-access/enroll-google-cloud-databases/spanner.mdx

```diff
@@ -0,0 +1,218 @@
+</TabItem>
+
+</Tabs>
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- Google Cloud account
+- Google Cloud Spanner database.
+- A host where you will run the Teleport Database Service, e.g. a Compute Engine
+  instance.
+- (!docs/pages/includes/tctl.mdx!)
+
+## Step 1/8. Create a service account for the Teleport Database Service
+
+Teleport uses [IAM authentication](https://cloud.google.com/spanner/docs/iam)
+to access the Cloud Spanner API.
+
+(!docs/pages/includes/database-access/cloudsql-create-service-account-for-db-service.mdx !)
+
+### Create a key for the service account
+
+(!docs/pages/includes/database-access/cloudsql_service_account_key.mdx!)
+
+## Step 2/8. Create a service account for a database user
+
+When a user connects to Spanner via Teleport, they will specify a service
+account name as the "database user" and the Teleport Database service will
+impersonate that service account to authenticate to GCP.
+
+You can create multiple service accounts to provide different access levels to
+Teleport users, but for this guide we will just create one. 
+
+(!docs/pages/includes/database-access/cloudsql_create_db_user_account.mdx!)
+
+### Grant permissions
+
+Click "select a role", choose the `Cloud Spanner Database User` role, and
+then click "Done":
+
+![Grant Cloud Spanner Database User to Service Account](../../../img/database-access/guides/cloudsql/grant-cloud-spanner-user@2x.png)
+
+<Admonition type="note">
+[Cloud Spanner Database User](https://cloud.google.com/spanner/docs/iam#spanner.databaseUser)
+is a pre-defined role,
+and the service account creation dialog we are using will bind permissions at the
+project level in GCP's
+[resource hierarchy](https://cloud.google.com/iam/docs/policies#inheritance).
+That means any allowed permissions will be granted to the service account for
+all Spanner instances within your GCP project.
+
+You can use a different pre-defined role or create and customize your own role
+permissions with
+[custom roles](https://cloud.google.com/spanner/docs/iam#custom-roles).
+You can also assign permissions at a more specific level in the resource
+hierarchy to limit the scope of the service account's permissions.
+
+Teleport's RBAC will limit the Spanner databases that Teleport users have
+access to, but we still recommend following the principal of least privilege -
+if only some of your Spanner databases will be accessed via Teleport, then limit
+the permission binding scope to just those databases.
+</Admonition>
+
+### Grant access to the service account
+
+The Teleport Database Service must be able to impersonate this service account.
+Navigate to the `teleport` service account overview page and select the
+"permissions" tab:
+
+![Select Service Account Permissions Tab](../../../img/database-access/guides/cloudsql/service-account-permissions-tab@2x.png)
+
+Click "Grant Access" and add the `teleport-db-service` principal ID.
```

Comment:
Similar to my last comment, would it make sense to show how to create a custom role with only the required permissions?

Reply:
I think it will be easier to follow the guide if we use pre-defined roles and just explain that custom roles are possible. The guide should get them to a working and reasonably safe state by default, and these pre-defined roles aren't *that* excessive imo.

The main thing is to not grant huge permission scope at the project level, because then the service account can impersonate every other service account in the project. This guide already restricts the token creator binding to just the `teleport` service account that `teleport-db-service` needs to impersonate.